docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,290 @@
+
+1. **Open Solution Explorer** in Visual Studio.
+2. **Right-click on the project** using MSTest and select **Properties**.
+3. Navigate to the **References** tab.
+4. Locate and **select the MSTest DLL references** (`Microsoft.VisualStudio.QualityTools.UnitTestFramework` and/or `Microsoft.VisualStudio.TestPlatform.TestFramework`).
+5. **Right-click** the selected reference(s) and choose **Remove**.
+
+For those comfortable editing XML directly, you can manually remove the references from your `.csproj` file as follows:
+
+```xml
+<Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+    <Private>False</Private>
+</Reference>
+```
+
+### 2. Update Your Project
+
+You can update your project to MSTest v3 in one of two ways:
+
+- **Install via NuGet Package**: Install the latest [MSTest](https://www.nuget.org/packages/MSTest) package using the NuGet Package Manager in Visual Studio or by running the following command in the NuGet Package Manager Console:
+
+    ```shell
+    Install-Package MSTest.TestFramework -Version 3.3.1
+    ```
+
+- **Or update the project file directly**: Update your `.csproj` file to specify the MSTest SDK version.
+
+    ```xml
+    <Project Sdk="MSTest.Sdk/3.3.1">
+      <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+      </PropertyGroup>
+    </Project>
+    ```
+
+Choose the option that best suits your project setup. Both methods ensure your project is upgraded to use MSTest v3.
+
+### 3. Update Your Code
+
+- **Replace Deprecated Methods**: Update deprecated methods to newer versions.
+  
+  - **Assert.AreEqual/AreNotEqual (with object)** → Use generic versions.
+
+    - **Before** (deprecated):
+
+    ```csharp
+    Assert.AreEqual(expectedObject, actualObject);
+    Assert.AreNotEqual(expectedObject, actualObject);
+    ```
+
+    - **After (using generics)**:
+
+    ```csharp
+    Assert.AreEqual<object>(expectedObject, actualObject);
+    Assert.AreNotEqual<object>(expectedObject, actualObject);
+    ```
+
+  - **Assert.AreSame/AreNotSame (with object)** → Use generic versions.
+
+    - **Before** (deprecated):
+
+    ```csharp
+    Assert.AreSame(expectedObject, actualObject);
+    Assert.AreNotSame(expectedObject, actualObject);
+    ```
+
+    - **After (using generics)**:
+
+    ```csharp
+    Assert.AreSame<object>(expectedObject, actualObject);
+    Assert.AreNotSame<object>(expectedObject, actualObject);
+    ```
+
+- **Test Initialization**: Use `TestInitialize` methods for async initialization.
+- **Cleanup**: Use `TestCleanup` methods or the `Dispose` pattern for cleanup.
+- **RunSettings**: The `.testsettings` file is no longer supported, meaning `<LegacySettings>` is also no longer available. Use [.runsettings](https://learn.microsoft.com/visualstudio/test/configure-unit-tests-by-using-a-dot-runsettings-file) for test configuration.
```

Comment:
We probably want to use the local link and not the full URL.

Reply:
it's in another repo

Reply:
You can, you should use something like that https://github.com/dotnet/docs/blob/main/docs/core/testing/unit-testing-mstest-configure.md?plain=1#L17